### ISSUE TITLE
G4 Doorbell Pro POE not streaming to HomeKit

### ISSUE THREAD

**Issue by tamorgen** [hjdhjd/homebridge-unifi-protect#1021]
### Homebridge UniFi Protect Version

homebridge-unifi-protect v6.16.0

### Homebridge Platform and OS

MacOS Sonoma 14.1

### Homebridge Version

v1.7.0

### Node Version

v18.16.0

### UniFi OS Version

3.1.16

### UniFi Protect Controller Version

Version 2.8.35

### Describe the problem

I just added a G4 Doorbell POE to my home.  The doorbell works perfectly however, the stream does not work in the Home app on my Mac.   It recognizes the camera, but when I click on the camera, there is a camera in the middle of the screen that is crossed out.  I've verified that the rtsps streams are enabled.  All of the other UniFi Protect cameras stream without issue.

### Homebridge HBUP JSON configuration

```json
{
            "controllers": [
                {
                    "address": "192.168.1.1",
                    "doorbellMessages": [
                        {
                            "message": "Welcome"
                        }
                    ],
                    "name": "Unifi Dream Machine Pro",
                    "password": "***********",
                    "username": "****************"
                }
            ],
            "platform": "UniFi Protect"
        }
```

### Relevant log output

```shell
[07/11/2023, 20:00:53] [UniFi Protect] UDM-Pro [UDM-PRO] G4 Doorbell Pro PoE Package Camera [G4 Doorbell Pro PoE]: Streaming request from 192.168.1.112: 1600x1200@30fps, 1,450 kbps. Using 1600x1200@2fps (Package Camera), 800 kbps.
[07/11/2023, 20:00:56] [UniFi Protect] UDM-Pro [UDM-PRO] G4 Doorbell Pro PoE Package Camera [G4 Doorbell Pro PoE]: Stopped video streaming session.
```

### Acknowledgment that you are only running UniFi OS and UniFi Protect releases from the Ubiquiti Official release channel

- [X] I acknowledge that I am not running any early access / beta firmwares or operating systems from either Ubiquiti or Apple.

**Comment by tamorgen** [comment]
Performed a reboot, got better logs:

`[08/11/2023, 06:50:46] Homebridge v1.7.0 (HAP v0.11.1) (Homebridge E51B) is running on port 51203.
[08/11/2023, 06:50:46] [UniFi Protect] Using FFmpeg version: 6.0-homebridge-alpine-armv7l-static.
(node:931) [DEP0123] DeprecationWarning: Setting the TLS ServerName to an IP address is not permitted by RFC 6066. This will be ignored in a future version.
(Use `node --trace-deprecation ...` to show where the warning was created)
[08/11/2023, 06:50:47] [UniFi Protect] UDM-Pro [UDM-PRO]: Connected to 192.168.1.1 (UniFi Protect 2.8.35 running on UniFi OS 3.1.16).
[08/11/2023, 06:50:47] [UniFi Protect] UDM-Pro [UDM-PRO]: Discovered nvr: UDM-Pro [UDM Pro] (address: 192.168.1.1 mac: 74ACB9DA1A89).
[08/11/2023, 06:50:47] [UniFi Protect] UDM-Pro [UDM-PRO]: Discovered camera: UVC G4 Dome - Front Yard [G4 Dome] (address: 192.168.1.21 mac: 68D79ACF695C).
[08/11/2023, 06:50:47] [UniFi Protect] UDM-Pro [UDM-PRO]: Discovered camera: UVC G4 Pro [G4 Pro] (address: 192.168.1.22 mac: 68D79ACF3E6D).
[08/11/2023, 06:50:47] [UniFi Protect] UDM-Pro [UDM-PRO]: Discovered camera: UVC G4 Dome - Driveway [G4 Dome] (address: 192.168.1.17 mac: 74ACB99E036F).
[08/11/2023, 06:50:47] [UniFi Protect] UDM-Pro [UDM-PRO]: Discovered camera: UVC G4 Dome Garage [G4 Dome] (address: 192.168.1.18 mac: 74ACB9002D80).
[08/11/2023, 06:50:47] [UniFi Protect] UDM-Pro [UDM-PRO]: Discovered camera: UVC G3 Flex Mia's Room [G3 Flex] (address: 192.168.1.19 mac: 74ACB90171ED).
[08/11/2023, 06:50:47] [UniFi Protect] UDM-Pro [UDM-PRO]: Discovered camera: UVC G3 Flex Mama Cave [G3 Flex] (address: 192.168.1.20 mac: 74ACB90171F8).
[08/11/2023, 06:50:47] [UniFi Protect] UDM-Pro [UDM-PRO]: Discovered camera: UVC-G4-Dome-Back-Right [G4 Dome] (address: 192.168.1.23 mac: 68D79ACFC614).
[08/11/2023, 06:50:47] [UniFi Protect] UDM-Pro [UDM-PRO]: Discovered camera: G4 Doorbell Pro PoE [G4 Doorbell Pro PoE] (address: 192.168.1.16 mac: F4E2C6735BEB).
[08/11/2023, 06:50:47] [UniFi Protect] UDM-Pro [UDM-PRO]: Discovered chime: UP Chime PoE [UP Chime PoE] (address: 192.168.1.24 mac: F4E2C67414AB).
[08/11/2023, 06:50:47] [UniFi Protect] UDM-Pro [UDM-PRO] G4 Doorbell Pro PoE [G4 Doorbell Pro PoE]: Enabled doorbell message switch: LEAVE PACKAGE AT DOOR.
[08/11/2023, 06:50:47] [UniFi Protect] UDM-Pro [UDM-PRO] G4 Doorbell Pro PoE [G4 Doorbell Pro PoE]: Enabled doorbell message switch: DO NOT DISTURB.
[08/11/2023, 06:50:47] [UniFi Protect] UDM-Pro [UDM-PRO] G4 Doorbell Pro PoE [G4 Doorbell Pro PoE]: Enabled doorbell message switch (60 seconds): Wilkommen.
[08/11/2023, 06:53:58] [UniFi Protect] UDM-Pro [UDM-PRO] G4 Doorbell Pro PoE Package Camera [G4 Doorbell Pro PoE]: Streaming request from 192.168.1.127: 640x480@30fps, 164 kbps. Using 1600x1200@2fps (Package Camera), 800 kbps.
[08/11/2023, 06:54:07] [UniFi Protect] UDM-Pro [UDM-PRO] G4 Doorbell Pro PoE Package Camera [G4 Doorbell Pro PoE]: Stopped video streaming session.
[08/11/2023, 06:54:09] [UniFi Protect] UDM-Pro [UDM-PRO] G4 Doorbell Pro PoE Package Camera [G4 Doorbell Pro PoE]: Streaming request from 192.168.1.127: 640x480@30fps, 164 kbps. Using 1600x1200@2fps (Package Camera), 800 kbps.
[08/11/2023, 06:54:14] [UniFi Protect] UDM-Pro [UDM-PRO] G4 Doorbell Pro PoE Package Camera [G4 Doorbell Pro PoE]: Stopped video streaming session.
`

**Comment by tamorgen** [comment]
Also of note, the package camera does stream from the doorbell, just not the main camera.

**Comment by hjdhjd** [comment]
I've got an identical setup without an issue. I know that's not much of a consolation...I'm at a bit of a loss here unfortunately beyond suggesting the usual reboot the doorbell, the Protect controller (your UDMP in this case), restart Homebridge, and your HomeKit hubs, and whatever you're trying to stream from (iPhone, etc.) in that order.

The request is being received by HBUP and HBUP is servicing it. Beyond that...there's not much for me/HBUP to do here...it's detecting the request and streaming properly. 

The HomeKit client isn't getting it somewhere along the way...that's usually either a network configuration issue, or HomeKit weirdness when adding a new camera. When it's the latter...a little bit of time and rebooting various things typically jostles HomeKit into submission.

Wish I could help more here...best of luck.

**Comment by czinda** [comment]
FWIW I had this very same issue.  I noticed my user for homekit needed more permissions.  I think this was unrelated because I had the package camera, just like you, but not the primary one.  Just felt I should mention it.

My fix was to reboot the UDM, delete HBUP, and added it all back.  Everything shows up as one would expect.

**Comment by tamorgen** [comment]
> FWIW I had this very same issue. I noticed my user for homekit needed more permissions. I think this was unrelated because I had the package camera, just like you, but not the primary one. Just felt I should mention it.
> 
> My fix was to reboot the UDM, delete HBUP, and added it all back. Everything shows up as one would expect.

So, I had done exactly as you did.  I rebooted my UDMP several times, I uninstalled the HBUP plugin from Homebridge and reinstalled, and reconfigured with the original configuration file.  I still wasn't getting a stream from the camera.   This morning I checked again, several days after the uninstallation/reinstallation, and the stream was now present.  I had checked yesterday, and it was still missing, so this somehow happened overnight.  I can't explain what happened, but the issue seems to have fixed itself.
